tests/utils/nnnunet_utils_test.py

```diff
@@ -0,0 +1,45 @@
+import logging
+import re
+
+import pytest
+from torch.optim import SGD
+
+from fl4health.utils.nnunet_utils import PolyLRSchedulerWrapper
+from tests.test_utils.models_for_test import MnistNetWithBnAndFrozen
+
+
+def test_poly_lr_scheduler(caplog: pytest.LogCaptureFixture) -> None:
+    max_steps = 1000
+    exponent = 1
+    steps_per_lr = 100
+    initial_lr = 0.5
+
+    model = MnistNetWithBnAndFrozen()
+    opt = SGD(model.parameters(), lr=initial_lr)
+    lr_scheduler = PolyLRSchedulerWrapper(
+        optimizer=opt, max_steps=max_steps, initial_lr=initial_lr, exponent=exponent, steps_per_lr=steps_per_lr
+    )
+
+    assert lr_scheduler.num_windows == round(max_steps / steps_per_lr)
+    assert lr_scheduler.initial_lr == initial_lr
+
+    prev_lr = initial_lr
+    for step in range(1, max_steps + 1):
+        curr_lr = lr_scheduler.get_lr()[0]
+
+        if step % steps_per_lr == 0:
+            assert curr_lr != prev_lr
+        else:
+            assert curr_lr == prev_lr
+
+        prev_lr = curr_lr
+
+        if step < max_steps:
```

Comment:
I might be wrong, but I think because of our loop bounds, this will always be true. So perhaps we just drop the if? If it's not always true, we can just adjust the range bounds so it is true and then the next step pops the warning and still drop the if I think

Reply:
Since the loop is until max_steps + 1, on the last iteration this condition will be False. I don't want to change bound down to max_steps because at the beginning of each iteration I get the current lr and check whether its correct (same or different as prev lr depending on step number). So I want to  perform this check on the last iteration but not perform the step as it would bring us to max_steps. For this, after the loop I perform an additional step and test that a warning is logged. Let me know if I am missing something or making a silly mistake :)

Reply:
Yup, I see. That works for me.